core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -35,6 +35,24 @@ package object config {
   private[spark] val SPARK_EXECUTOR_PREFIX = "spark.executor"
   private[spark] val SPARK_TASK_PREFIX = "spark.task"
 
+  private[spark] val SPARK_RESOURCES_COORDINATE =
+    ConfigBuilder("spark.resources.coordinate.enable")
+      .doc("Whether to coordinate resources automatically among workers/drivers(client only) " +
+        "in Standalone. If not, user should be responsible for assigning different resources " +
+        "for workers/drivers while using resource discovery script.")
+      .booleanConf
+      .createWithDefault(true)
+
+  private[spark] val SPARK_RESOURCES_DIR =
+    ConfigBuilder("spark.resources.dir")
+      .doc("Directory used to coordinate resources among workers/drivers(client only) in " +
+        "Standalone. Default is SPARK_HOME. Make sure to use the same directory with the " +
+        "worker when submits the application in client mode and don't clean up this " +
```

Comment:
Make sure to use the same directory for worker and drivers in client mode that run on the same host.  Don't clean...